draft-ietf-privacypass-rate-limit-tokens.md

```diff
@@ -922,6 +957,40 @@ struct {
 } Token;
 ~~~
 
+## Penalization of Invalid Clients and Issuers {#penalization}
+
+When an Attester encounters incorrect and potentially malicious behavior
+by either Clients or Issuers, it needs to "penalize" them. This involves
+keeping state about invalid protocol events, and determining when a
+particular Client or Issuer has exceeded a threshold for such events.
+
+Each category of invalid events can have a different threshold, and can apply
+to Clients, Issuers, or both. Different penalization events are described
+below, with suggested thresholds. This list is not exhaustive.
+
+- Client Key changing more than once over any two consecutive policy windows.
+This is a Client-specific penalization event. A suggested threshold for
+penalizing the Client based on this is one event.
+
+- Issuer responses missing the "Sec-Token-Origin" header on a 2xx response.
+This is an Issuer-specific penalization event. A suggested threshold for
+penalizing the Issuer based on this is ten events across all Clients.
```

Comment:
Might be useful to have some rationalization for the suggested thresholds, since some of these seem slightly arbitary, or at least a way for attesters to make their own determination about acceptable thresholds.

Reply:
Added!